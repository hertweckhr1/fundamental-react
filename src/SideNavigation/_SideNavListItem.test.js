import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import SideNavList from './_SideNavList';
import SideNavListItem from './_SideNavListItem';
import { BrowserRouter, Link } from 'react-router-dom';

describe('<SideNavListItem />', () => {
    const oneLevelSideNav = (
        <BrowserRouter>
            <SideNavList>
                <SideNavListItem
                    id='item_1'>
                    <Link to='/'>
                            Link Item
                    </Link>
                </SideNavListItem>
                <SideNavListItem
                    id='item_2'>
                    <Link to='/'>
                            Link Item
                    </Link>
                </SideNavListItem>
                <SideNavListItem
                    id='item_3'>
                    <Link to='/'>
                            Link Item
                    </Link>
                </SideNavListItem>
                <SideNavListItem
                    id='item_4'>
                    <Link to='/'>
                            Link Item
                    </Link>
                </SideNavListItem>
                <SideNavListItem
                    id='item_5'>
                    <Link to='/'>
                            Link Item
                    </Link>
                </SideNavListItem>
            </SideNavList>
        </BrowserRouter>
    );

    test('create side navigation', () => {
        // create one level side nav
        let component = renderer.create(oneLevelSideNav);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the SideNavListItem component', () => {
            const element = mount(
                <SideNavListItem
                    data-sample='Sample'
                    id='item-1'
                    name='Link Item 1'
                    url='#' />
            );

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

    });
});
