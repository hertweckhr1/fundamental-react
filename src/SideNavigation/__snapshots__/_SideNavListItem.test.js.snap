// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNavListItem /> create side navigation 1`] = `
<ul
  className="fd-side-nav__list"
>
  <li
    className="fd-side-nav__item"
  >
    <a
      className="fd-side-nav__link"
      href="/"
      onClick={[Function]}
    >
      Link Item
    </a>
  </li>
  <li
    className="fd-side-nav__item"
  >
    <a
      className="fd-side-nav__link"
      href="/"
      onClick={[Function]}
    >
      Link Item
    </a>
  </li>
  <li
    className="fd-side-nav__item"
  >
    <a
      className="fd-side-nav__link"
      href="/"
      onClick={[Function]}
    >
      Link Item
    </a>
  </li>
  <li
    className="fd-side-nav__item"
  >
    <a
      className="fd-side-nav__link"
      href="/"
      onClick={[Function]}
    >
      Link Item
    </a>
  </li>
  <li
    className="fd-side-nav__item"
  >
    <a
      className="fd-side-nav__link"
      href="/"
      onClick={[Function]}
    >
      Link Item
    </a>
  </li>
</ul>
`;
